src/api/routes/rosetta/construction.ts

```diff
@@ -536,10 +536,7 @@ export function createRosettaConstructionRouter(db: DataStore, chainId: ChainID)
     }
 
     if (has0xPrefix(signatures[0].public_key.hex_bytes)) {
-      signatures[0].signing_payload.hex_bytes = signatures[0].signing_payload.hex_bytes.replace(
-        '0x',
-        ''
-      );
+      signatures[0].public_key.hex_bytes = signatures[0].public_key.hex_bytes.replace('0x', '');
```

Comment:
```suggestion
      signatures[0].public_key.hex_bytes = signatures[0].public_key.hex_bytes.slice(2);
```

This is more accurate, and avoids casing issues like `0XABCD`.